### DISCUSSION TITLE
the "${staged}" arg only returns empty

### DISCUSSION THREAD

**Discussion by foxmcloud555** [alirezanet/Husky.Net#91] [Q&A] [answered]
Just setting up Husky across a team project right now, and I'm apparently struggling to use the args in the task runner. 

I have doubled checked both with and without staged files to see if anything changed, it didn't make a difference. 

Here's my task-runner.json:

<img width="579" alt="image" src="https://github.com/alirezanet/Husky.Net/assets/20091585/aa1b9554-7919-4752-8b2c-fc71583a4605">

And here's my terminal output on `dotnet husky run `

<img width="564" alt="image" src="https://github.com/alirezanet/Husky.Net/assets/20091585/5d578caa-fdd9-4874-8600-823f9b640aef">

If I don't use the --include, inspectcode runs as normal and the output is as expected. 

I am on an M1 mac, I know some dotnet tools are currently a bit funky with arm arch but everything aside from this seems fine so far. 

There is a submodule present, but currently i'm in the parent repo. 

I've tried with a fresh clone of this repo in a new location just to make sure, and I've had a look through and nothing looks out of the ordinary. Hooks are triggered as you'd expect, so the pre-commit group do actually run on commit, etc. 

It's just this specific variable thats always empty.

I would greatly appreciate any help.

**Comment by alirezanet** [comment]
Hi @foxmcloud555,
Do you have access to a Windows or Linux machine? since I don't have a mac, I can not debug it. just want to make sure whether this is OS related issue or we have a bug somewhere ... 
also please check what is the output of this command for you? 
``` bash
git diff --staged --name-only --no-ext-diff --diff-filter=AM
```

I did test this task and it works as expected:
``` json
     {
         "name": "echo staged files",
         "pathMode": "absolute",
         "command": "cmd",
         "group": "pre-commit",
         "args": [ "/c", "echo", "${staged}"],
      }
```
pre-commit:
```
#!/bin/sh
. "$(dirname "$0")/_/husky.sh"

dotnet husky run -v --group "pre-commit"
```

**Comment by foxmcloud555** [comment]
Thanks for looking at this!

I just got set up on a windows device to check for you, and everything works just fine. 

It echos the staged files with no issue. 

Also, the output of that git command is exactly as you'd expect on both devices, mac and windows. 

I'll clone the actual husky repo today and see if I can run through.

**Reply by alirezanet** [reply]
It would be nice if you could look into this, for this library I think I have to buy a Mac machine 😅

**Comment by alirezanet** [comment, marked as answer]
Today, I encountered the same issue on my Linux machine. To echo staged files in Linux, and likely on Mac as well, you should define the task as follows:

```json
      {
             "tasks": [
                 {
                     "name": "Test",
                     "command": "echo",
                     "args": [ "${staged}" ]
                 }
             ]
         }
```

So basically we don't need `bash -c`
